src/Middleware/ForceSecureConnection.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Yii\Web\Middleware;
+
+use Psr\Http\Message\ResponseFactoryInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\MiddlewareInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+use Yiisoft\Http\Status;
+
+/**
+ * Redirects from HTTP to HTTPS and adds CSP and HSTS headers
```

Comment:
A note should be added that it's better to do it with webserver config. See https://github.com/yiisoft/yii-web/issues/237